network/gopubsub/pubsub.go

```diff
@@ -0,0 +1,126 @@
+package gopubsub
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+)
+
+//PubSubChannel represents a pubsub system where subscriber can .Subscribe() and publishers can .Publish() or .Close().
+type PubSubChannel struct {
+	subscriptions []*Subscription
+	subsMutex     sync.RWMutex
+	nextId        int
+}
+
+//Subscription is created in PubSubChannel using pubSub.Subscribe(). Subscribers can receive using .ReceiveChannel().
```

Comment:
Comments should be separated with spaces from //, like in most of the packages in this repo.